src/new_index/mempool.rs

```diff
@@ -520,6 +490,46 @@ impl Mempool {
             .get(asset_id)
             .map_or_else(|| vec![], |entries| self._history(entries, limit))
     }
+
+    pub fn update(mempool: &Arc<RwLock<Mempool>>, daemon: &Daemon) -> Result<()> {
+        let _timer = mempool.read().unwrap().latency.with_label_values(&["update"]).start_timer();
+
+        // 1. Determine which transactions are no longer in the daemon's mempool and which ones have newly entered it
+        let old_txids = mempool.read().unwrap().old_txids();
+        let new_txids = daemon
+            .getmempooltxids()
+            .chain_err(|| "failed to update mempool from daemon")?;
+        let to_remove: HashSet<&Txid> = old_txids.difference(&new_txids).collect();
+
+        // 2. Remove missing transactions. Even if we are unable to download new transactions from
+        // the daemon, we still want to remove the transactions that are no longer in the mempool.
+        mempool.write().unwrap().remove(to_remove);
+
+        // 2. Download the new transactions from the daemon's mempool
+        let txids: Vec<&Txid> = new_txids.difference(&old_txids).collect();
```

Comment:
Maybe give `txids` a more descriptive name? Previously this was part of a separate method, but here we have a bunch of different variables with txids and txs which can get confusing

Reply:
In fact, I think it would be good if we changed all the txids/txs variable names to be more consistent. Perhaps `old_txids`, `new_txids`, `removed_txids`, `added_txids` and `added_txs`?

Reply:
Fixed! I've tried to make them as descriptive as possible. Please let me know what you think.